arch/xtensa/soc/esp32/linker.ld

```diff
@@ -38,7 +39,7 @@ MEMORY
 {
   iram0_0_seg(RX): org = 0x40080000, len = 0x20000
   iram0_2_seg(RX): org = 0x400D0018, len = 0x330000
-  dram0_0_seg(RW): org = 0x3FFB0000, len = 0x50000
+  dram0_0_seg(RW): org = CONFIG_SRAM_BASE_ADDRESS, len = CONFIG_SRAM_SIZE*1K
```

Comment:
Does this bit have anything to do with DTS?  It seems dangerous to me.  I know we have those kconfig tunables, and we surely hope they're correct, but these segment addresses are the actual, true memory map of the device and can't be reasonably changed.  Allowing this to be set externally is just a footgun, no?

Reply:
@andyross  as a part of dts migration for all boards, we are deriving ram address from .dtsi file . In this case it is "dts/xtensa/esp32.dtsi". Where "sram0" node represent the memory mapped address of ram and it is specific to Soc.  This is the way it is done for all other arch based boards.

There is no One kibitz of memory mapping change. The dts script will generate CONFIG_SRAM_SIZE to 320. The size will be multiple of Kb.